pyasic/miners/backends/auradine.py

```diff
@@ -193,6 +195,53 @@ async def send_config(self, config: MinerConfig, user_suffix: str = None) -> Non
         for key in conf.keys():
             await self.web.send_command(command=key, **conf[key])
 
+    async def upgrade_firmware(self, *, url: str = None, file_path: str = None, version: str = "latest") -> bool:
+        """
+        Upgrade the firmware of the Auradine device.
+
+        Args:
+            url (str): The URL to download the firmware from.
+            file_path (str): The local file path of the firmware to be uploaded.
+            version (str): The version of the firmware to upgrade to.
+
+        Returns:
+            bool: True if the firmware upgrade was successful, False otherwise.
+        """
+        try:
+            logging.info("Starting firmware upgrade process.")
+
+            if not url and not file_path:
+                raise ValueError("Either URL or file path must be provided for firmware upgrade.")
+
+            if url:
+                result = await self.web.firmware_upgrade(url=url)
+            elif file_path:
```

Comment:
File path should just be ignored for these machines I think, there might be a way to spin up a server or something in the future, but these miners can't be upgraded this way.